exporters/zipkin/src/main/java/io/opentelemetry/exporter/zipkin/OtelToZipkinSpanTransformer.java

```diff
@@ -0,0 +1,233 @@
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.sdk.common.InstrumentationScopeInfo;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.sdk.trace.data.EventData;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.sdk.trace.data.StatusData;
+import io.opentelemetry.semconv.resource.attributes.ResourceAttributes;
+import java.net.InetAddress;
+import java.net.NetworkInterface;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+import zipkin2.Endpoint;
+import zipkin2.Span;
+
+/**
+ * This class is responsible for transforming an OpenTelemetry SpanData instance into an instance of
+ * a Zipkin Span. It is based, in part, on code from
+ * https://github.com/census-instrumentation/opencensus-java/tree/c960b19889de5e4a7b25f90919d28b066590d4f0/exporters/trace/zipkin
+ */
+public final class OtelToZipkinSpanTransformer implements Function<SpanData, Span> {
+
+  static final String KEY_INSTRUMENTATION_SCOPE_NAME = "otel.scope.name";
+  static final String KEY_INSTRUMENTATION_SCOPE_VERSION = "otel.scope.version";
+  static final String KEY_INSTRUMENTATION_LIBRARY_NAME = "otel.library.name";
+  static final String KEY_INSTRUMENTATION_LIBRARY_VERSION = "otel.library.version";
+  static final String OTEL_DROPPED_ATTRIBUTES_COUNT = "otel.dropped_attributes_count";
+  static final String OTEL_DROPPED_EVENTS_COUNT = "otel.dropped_events_count";
+  static final String OTEL_STATUS_CODE = "otel.status_code";
+  private static final Logger logger = Logger.getLogger(ZipkinSpanExporter.class.getName());
+  static final AttributeKey<String> STATUS_ERROR = stringKey("error");
+  private final Supplier<Optional<InetAddress>> ipAddressSupplier;
+
+  /**
+   * Creates a new instance of an OtelToZipkinSpanTransformer. This version of the constructor will
+   * use a fixed IP address that is fetched from the network interfaces at construction time.
+   */
+  public OtelToZipkinSpanTransformer() {
+    Optional<InetAddress> inetAddress = produceLocalIp();
+    this.ipAddressSupplier = () -> inetAddress;
+  }
+
+  /**
+   * Creates an instance of an OtelToZipkinSpanTransformer with the given Supplier that can produce
+   * an optional InetAddress. This value from this Supplier will be used when creating the local
+   * zipkin Endpoint for each Span.
+   *
+   * @param ipAddressSupplier - A Supplier of an Optional InetAddress
+   */
+  public OtelToZipkinSpanTransformer(Supplier<Optional<InetAddress>> ipAddressSupplier) {
+    this.ipAddressSupplier = ipAddressSupplier;
+  }
+
+  @Override
+  public Span apply(SpanData spanData) {
+    return generateSpan(spanData);
+  }
+
+  Span generateSpan(SpanData spanData) {
```

Comment:
If you want this to implement `Function<SpanData, Span>` then its silly to give the `apply` method an alias for internal testing. Just call `apply`. However, `OtelToZipkinSpanTransformer` doesn't need to explicitly implement `Function` in order to qualify as one. I.e. you can get rid of `implements Function` and still declare `private Function<SpanData, Span> otelToZipkinTransformer = new OtelToZipkinSpanTransformer()::generateSpan` in `ZipkinSpanExporterBuilder`. 

A disadvantage of using `Function` in general is that we don't get proper nullaway support with it. The response of `apply` can be `null`, but the static code checking doesn't know that and we can get mysterious NPEs.

Reply:
Yeah, the `generateSpan` function is there I think because that's what the method was called originally, and the `Function` interface came on after. I thought it would be more flexible (you know, pass and declare by interface and all).

Maybe it doesn't buy us much. I'll look at stripping `Function` interface from the class.